### DISCUSSION TITLE
Class cannot subclass "Screen", "Message", etc.

### DISCUSSION THREAD

**Discussion by PabloLec** [Textualize/textual#2429] [Q&A] [answered]
Hi,

MyPy is raising the following exceptions:
```
Class cannot subclass "ListView" (has type "Any")  [misc]
Class cannot subclass "Screen" (has type "Any")  [misc]
Class cannot subclass "Message" (has type "Any")  [misc]
Class cannot subclass "Screen" (has type "Any")  [misc]
Class cannot subclass "Horizontal" (has type "Any")  [misc]
```

In my code, I'm subclassing in the most regular way, e.g for Screen : 
```
class ModalScreen(Screen):
    ...
```

And get the previous exceptions.
It started happening when upgrading textual from 0.20.1 to 0.21.0.
Any idea why ? I'm clueless about this MyPy error and would be curious to know its reason.

**Comment by davep** [comment, marked as answer]
I would expect mypy to perhaps give a warning about inheriting from `Screen` with more recent versions of Textual as there is a new feature -- still to be properly documented -- for typing a screen and being able to return a result from a screen when it's dismissed. See https://github.com/Textualize/textual/pull/2321 for more details on that.

If you're not intending to return a result from your modal screen you can change the code you show above to be like this:

```python
class ModalScreen(Screen[None]):
    ...
```

As a slight aside to this, you might want to note that [`ModalScreen` is now a type in Textual](https://textual.textualize.io/api/screen/#textual.screen.ModalScreen).

More generally though, I'm not clear on why you'd be seeing the warnings you're showing there. I'm running `mypy` v1.2.0 here and testing it on a couple of my own projects that are heavy on inheriting from things in Textual I'm not seeing anything like that.

Do you think you could show a simple stand-alone bit of code that demonstrates these warnings, and also give details of your `mypy` version, plus your Python version, so I can test the same here?

**Reply by PabloLec** [reply]
Thanks, @davep, it did the trick! So, the type passed to Screen is its return type. And is it correct that the use of `Generic` makes it mandatory to type hint? Additionally, is the advantage of type hinting the `ScreenResultType` to provide a hint for the eventual return type of its callback?

By the way, I'm also using mypy 1.2, but with the --strict flag. I encountered this error for Screen and App. I'm not sure why I saw this error for Horizontal, etc. before, but I'm not seeing it today, which is weird but not textual related."

**Reply by davep** [reply]
> Additionally, is the advantage of type hinting the ScreenResultType to provide a hint for the eventual return type of its callback?

Spot on, exactly this.

Yup, I always run with the `--strict` flag too.
